src/login/loginctl.c

```diff
@@ -77,6 +77,37 @@ typedef struct SessionStatusInfo {
         dual_timestamp idle_hint_timestamp;
 } SessionStatusInfo;
 
+typedef struct UserStatusInfo {
+        uid_t uid;
+        bool linger;
+        const char *name;
+        struct dual_timestamp timestamp;
+        const char *state;
+        char **sessions;
+        const char *display;
+        const char *slice;
+} UserStatusInfo;
+
+typedef struct SeatStatusInfo {
+        const char *id;
+        const char *active_session;
+        char **sessions;
+} SeatStatusInfo;
+
+static void user_status_info_clear(UserStatusInfo *info) {
+        if (info) {
+                strv_free(info->sessions);
+                zero(*info);
+        }
+}
```

Comment:
It is not necessary to clear the struct with zero. Hence, please rename it as `user_status_info_done()`.
Also, unlike `xyz_free()`, the function argument here should not be `NULL`. So, please do something like the following:
```c
static void user_status_info_done(UserStatusInfo *info) {
        assert(info);

        strv_free(info->sessions);
}
```

(I know this is not your fault, but when you touch the code, please follow our new coding style.)